### DISCUSSION TITLE
docker-slim failing to include dependencies

### DISCUSSION THREAD

**Discussion by davidgr-pge** [slimtoolkit/slim#333] [Q&A]
hi folks. I'm new to docker-slim and trying to minify a python app, but I can't seem to get the necessary dependencies to execute the container's entrypoint.

I'm having trouble using the http probing, so I thought that including the entrypoint as a --exec command would help docker-slim to find the dependencies for it. Is that not the case? Obviously it hasn't worked for me yet.

Here's my command and output

```docker-slim build --target model:local --http-probe-off --exec "seldon-core-microservice"```
```cmd=build info=continue.after mode='exec' message='provide the expected input to allow the container inspector to continue its execution'
cmd=build info=continue.after mode='exec' shell='seldon-core-microservice'
docker-slim[build][exec]: output: Traceback (most recent call last):
docker-slim[build][exec]: output:   File "/app/.local/bin/seldon-core-microservice", line 5, in <module>
docker-slim[build][exec]: output:     from seldon_core.microservice import main
docker-slim[build][exec]: output: ModuleNotFoundError: No module named 'seldon_core'
```

what am I missing here?

docker-slim version darwin|Transformer|1.37.5|latest|latest

**Comment by kcq** [comment]
Is it supposed to be possible to run `seldon-core-microservice` directly? It's possible that the `exec` call needs additional environment variables... Where is the 'seldon_core' module located?

One thing to try is the --exec-file parameter. Try creating a shell script that calls seldon-core-microservice and point the --exec-file to that file. If it doesn't work as-is try setting `PYTHONPATH` in that shell script explicitly to point to the locations where the "missing" modules are located.

What do the ENTRYPOINT and CMD instructions look like in your Dockerfile, by the way?

What kind of microservice do you have? Is it an HTTP service? What kind of framework do you use? Flask? Django?
